test/e2e/framework/framework.go

```diff
@@ -206,16 +117,68 @@ func KubernetesClientEnv(t *testing.T) []string {
 	return []string{"KUBERNETES_SERVICE_HOST=" + host, "KUBERNETES_SERVICE_PORT=" + port}
 }
 
+type Accessory interface {
+	Ready()
+	Bind(cmd CiOperatorCommand)
+}
+
+var _ Accessory = &accessory{}
+
+type accessory struct {
```

Comment:
Hiding the implementation behind an interface like this doesn't really provide any value when there is only one implementation and makes navigating the code much harder

Reply:
(You do this in a variety of places in this PR. Generally it is recommended to construct concrete implementations and define the interface at the consumer side which doesn't really make much sense here so should be omitted)

Reply:
I guess I'm a proponent of not letting people do the wrong thing unless they really mean it. I don't want to expose this since you can get into really hairy, irritating deadlocks and timeouts if you mess with things ...

Reply:
That makes sense, but you already get that by having put these types into a distinct package and only exporting a subset of their fields and methods